src/main/java/use_case/CreateComment/CreateCommentInteractor.java

```diff
@@ -0,0 +1,81 @@
+package use_case.CreateComment;
+
+import entity.ModeratorUser;
+import entity.NormalComment;
+import entity.User;
+
+/**
+ * The Creation Comment Interactor.
+ */
+public class CreateCommentInteractor implements CreateCommentInputBoundary {
+    private final CreateCommentDataAccessInterface commentDataAccessObject;
+    private final CreateCommentOutputBoundary commentPresenter;
+
+    public CreateCommentInteractor(CreateCommentDataAccessInterface commentDataAccessObject,
+                                   CreateCommentOutputBoundary commentPresenter) {
+        this.commentDataAccessObject = commentDataAccessObject;
+        this.commentPresenter = commentPresenter;
+    }
+
+    @Override
+    public void execute(CreateCommentInputData createCommentInputData) {
+        if (createCommentInputData.getContent() == null || createCommentInputData.getContent().isEmpty()) {
+            commentPresenter.prepareFailView("Content cannot be empty.");
+            return;
+        }
+
+        if (createCommentInputData.getAuthor() == null) {
+            commentPresenter.prepareFailView("Author cannot be empty.");
+            return;
+        }
+
+        // THIS WILL HAVE TO BE CHANGED LATER BUT FOR NOW IT WILL BE LIKE THIS
+        if (createCommentInputData.getCommentId() == null ) {
+            commentPresenter.prepareFailView("Comment ID cannot be empty.");
+            return;
+        }
+
+        if (createCommentInputData.getEntryId() == null || createCommentInputData.getEntryId().isEmpty()) {
+            commentPresenter.prepareFailView("Entry ID cannot be empty.");
+            return;
+        }
+
+        if (createCommentInputData.getModerator() == null) {
+            commentPresenter.prepareFailView("Moderator cannot be empty.");
+            return;
+        }
+
+        // THIS WILL HAVE TO BE CHANGED LATER BUT FOR NOW IT WILL BE LIKE THIS
+        // talking about getCommentId()
+        NormalComment comment = new NormalComment(
```

Comment:
We should not instantiate a comment object here, use CreateCommentOutputData object